src/Sulu/Bundle/AdminBundle/Controller/IconController.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\AdminBundle\Controller;
+
+use FOS\RestBundle\View\ViewHandlerInterface;
+use HandcraftedInTheAlps\RestRoutingBundle\Routing\ClassResourceInterface;
+use Sulu\Component\Rest\AbstractRestController;
+use Sulu\Component\Rest\ListBuilder\CollectionRepresentation;
+use Sulu\Component\Rest\RequestParametersTrait;
+use Symfony\Component\Cache\Adapter\FilesystemAdapter;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\HttpFoundation\Exception\BadRequestException;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class IconController extends AbstractRestController implements ClassResourceInterface
+{
+    use RequestParametersTrait;
+
+    /**
+     * @var array<string, string>
+     */
+    private array $iconSets;
+
+    private FilesystemAdapter $cache;
+
+    /**
+     * @param array<string, string> $iconSets
+     */
+    public function __construct(
+        ViewHandlerInterface $viewHandler,
+        array $iconSets,
+    ) {
+        parent::__construct($viewHandler);
+        $this->iconSets = $iconSets;
+        $this->cache = new FilesystemAdapter();
+    }
+
+    /**
+     * Returns icons.
+     *
+     * @return Response
+     */
+    public function cgetAction(Request $request)
+    {
+        /** @var string $iconSetName */
+        $iconSetName = $this->getRequestParameter($request, 'icon_set', true);
+        $iconSet = \explode('://', $this->iconSets[$iconSetName]);
+        $search = $request->query->get('search');
+        $provider = $iconSet[0];
+        $path = $iconSet[1] ?? '';
+
+        // Get icons depending on the provider type.
+        switch ($provider) {
+            case 'svg':
```

Comment:
Extending this or adding custom providers always requires you to override the `IconController`, would be nice to have the providers as separate services, so that you just have to create a custom class, implementing the provider interface